WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -105,3 +62,99 @@ extension BlogListView.Site {
         )
     }
 }
+
+extension BlogListViewModel: NSFetchedResultsControllerDelegate {
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        if controller == pinnedSitesController {
+            pinnedSites = (controller.fetchedObjects as? [Blog] ?? []).compactMap(BlogListView.Site.init)
+        } else if controller == recentSitesController {
+            recentSites = (controller.fetchedObjects as? [Blog] ?? []).compactMap(BlogListView.Site.init)
+        } else if controller == allRemainingSitesController {
+            allRemainingSites = (controller.fetchedObjects as? [Blog] ?? []).compactMap(BlogListView.Site.init)
+        } else if controller == allBlogsController {
+            allBlogs = controller.fetchedObjects as? [Blog] ?? []
+            searchSites = allBlogs.compactMap(BlogListView.Site.init)
+        }
+    }
+}
+
+extension BlogListViewModel {
+    private func setupFetchedResultsControllers() {
+        pinnedSitesController = createResultsController(
+            with: NSPredicate(format: "pinnedDate != nil"),
+            descriptor: NSSortDescriptor(key: "pinnedDate", ascending: false)
+        )
+        recentSitesController = createResultsController(
+            with: NSPredicate(format: "lastUsed != nil AND pinnedDate == nil"),
+            descriptor: NSSortDescriptor(key: "lastUsed", ascending: false),
+            fetchLimit: 8
+        )
+        allRemainingSitesController = createResultsController(
+            with: NSPredicate(format: "lastUsed == nil AND pinnedDate == nil"),
```

Comment:
If I understand this correctly, if recent sites are more than 8, some sites won't show in either section.

Reply:
Hmm that's true. Good point!

Reply:
Resolved in [89da7b1](https://github.com/wordpress-mobile/WordPress-iOS/pull/22833/commits/89da7b1a0740fd381b42fe73eeb6a10901f1d5bd)